code/game/machinery/deployable.dm

```diff
@@ -210,6 +218,167 @@
+	qdel(src)
+
+/obj/structure/dropwall_generator
+	name = "deployed dropwall shield generator"
+	desc = "This generator designed by Shellguard Munitions's spartan division is used to deploy a temporary cover that blocks projectiles and explosions from a direction, while allowing projectiles to pass freely from behind."
+	icon = 'icons/obj/dropwall.dmi'
+	icon_state = "dropwall_deployed"
+	max_integrity = 25 // 2 shots
+	var/list/connected_shields = list()
+	/// This variable is used to prevent damage to it's core shield when it is up.
+	var/protected = FALSE
+	///The core shield that protects the generator
+	var/obj/structure/barricade/dropwall/core_shield = null
+
+/obj/structure/dropwall_generator/Initialize(mapload, direction, uptime)
+	. = ..()
+	if(direction)
+		deploy(direction, uptime)
+
+/obj/structure/dropwall_generator/proc/deploy(direction, uptime)
+	anchored = TRUE
+	protected = TRUE
+	addtimer(CALLBACK(src, .proc/power_out), uptime)
+	while(uptime >= 10)
+		addtimer(CALLBACK(src, .proc/annoying_overlay_proc, uptime / 10), (10 * (12 - uptime / 10))) // instantly, 1 second, 2 seconds, ect. Would put in SECONDS here but it shits itself
+		uptime -= 10
+
+	connected_shields += new /obj/structure/barricade/dropwall(get_turf(loc), src, TRUE, direction)
+	for(var/i in connected_shields)
+		core_shield = i
+
+	var/dir_left = turn(direction, -90)
+	var/dir_right = turn(direction, 90)
+	var/target_turf = get_step(src, dir_left)
+	if(!(is_blocked_turf(target_turf)))
+		connected_shields += new /obj/structure/barricade/dropwall(target_turf, src, FALSE, direction, dir_left)
+
+	var/target_turf2 = get_step(src, dir_right)
+	if(!(is_blocked_turf(target_turf2)))
+		connected_shields += new /obj/structure/barricade/dropwall(target_turf2, src, FALSE, direction, dir_right)
+
+
+/obj/structure/dropwall_generator/attacked_by(obj/item/I, mob/living/user) //No, you can not just go up to the generator and whack it. Central shield needs to go down first.
+	if(protected)
+		visible_message("<span class='warning'>[src]'s shield absorbs the blow!</span>")
+		core_shield.take_damage(I.force, I.damtype, MELEE, 1)
+	else
+		. = ..()
+
+/obj/structure/dropwall_generator/bullet_act(obj/item/projectile/P)
+	if(protected)
+		visible_message("<span class='warning'>[src]'s shield absorbs the blow!</span>")
+		core_shield.take_damage(P.damage, P.damage_type, P.flag)
+	else
+		. = ..()
+
+/obj/structure/dropwall_generator/Destroy()
+	QDEL_LIST(connected_shields)
+	core_shield = null
+	. = ..()
+
+/obj/structure/dropwall_generator/emp_act(severity)
+	..()
+	if(protected)
+		for(var/obj/structure/barricade/dropwall/O in connected_shields)
+			O.emp_act(severity)
+	else
+		qdel(src)
+
+/obj/structure/dropwall_generator/proc/power_out()
+	visible_message("<span class='warning'>[src] runs out of power, causing its shields to fail!</span>")
+	new /obj/item/used_dropwall(get_turf(src))
+	qdel(src)
+
+/obj/structure/dropwall_generator/proc/annoying_overlay_proc(uptime)
```

Comment:
Renaming this is prob also best to explain what the proc does

Reply:
Done